File: 2015/5/Nice.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

public class Nice {

    public static void main(String[] args) {

        int counter = 0;

        try (BufferedReader file = new BufferedReader(new FileReader("input.txt"))) {

            String line = null;
            while((line = file.readLine()) != null) {
                if (isNice(line)) {
                    counter++;
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        System.out.println(counter);
    }

    public static boolean isNice(String input) {

        boolean flag_1 = false;
        boolean flag_2 = false;

        ArrayList<String> string = new ArrayList<>();
        for (String character : input.split("")) {
            string.add(character);
        }
        
        for (int i = 0; i < (string.size() - 2); i++) {
            if ((string.get(i)).equals(string.get(i+2))) {
                flag_1 = true;
                break;
            }
        }
        
        for (int i = 0; i < (string.size() - 1); i++) {
            ArrayList<String> array = string;
            String char_1 = array.get(0);
            String char_2 = array.get(1);
            array.remove(0);
            array.remove(1);

            for (int n = 0; n < (array.size() - 1); n++) {
                if ((char_1 + char_2).equals(array.get(n) + array.get(n+1))) {
                    flag_2 = true;
                    break;
                }
            }
        }

        if (flag_1 && flag_2) {
            return true;
        } else {
            return false;
        }
    }
}

File: 2015/5/nice.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

std::vector<char> str_to_vec(std::string line)
{
    std::vector<char> letters;

    for (char c : line)
    {
        letters.push_back(c);
    }

    return letters;
}

std::vector<std::vector<char>> file_to_vec(const char *path)
{
    std::vector<std::vector<char>> file;

    std::ifstream input;
    input.open(path);

    if (!input)
    {
        std::cout << "Error during input file opening" << std::endl;
        return {};
    }
    
    std::string line;

    getline(input, line);
    while (!input.eof())
    {
        file.push_back(str_to_vec(line));
        getline(input, line);
    }

    return file;
}

bool is_nice(std::vector<char> word)
{
    bool flag_1 = false;
    bool flag_2 = false;

    for (int i = 0; (i < word.size() - 2) && (flag_1 == false); i++)
    {
        if (word[i] == word[i+2]) {flag_1 = true;}
    }

    for (int i = 0; (i < word.size() - 3) && (flag_2 == false); i++)
    {
        for (int j = i+2; (j < word.size() - 1) && (flag_2 == false); j++)
        {
            if ((word[i] == word[j]) && (word[i+1] == word[j+1])) {flag_2 = true;}
        }
    }

    return (flag_1 && flag_2);
}

int main()
{
    std::vector<std::vector<char>> input = file_to_vec("input.txt");

    int counter = 0;
    for (std::vector<char> line : input)
    {
        if (is_nice(line))
        {
            counter++;
        }
    }

    std::cout << counter << std::endl;

    return 0;
}